### PR TITLE
Updated Tests for Team to Mock Database.

### PR DESCRIPTION
**Pull request by OlinZ**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

## Description of Changes

Updated `team.test.ts` to mock database functions. Need some feedback since we are not sure if we went in the right direction. 

## Related Issues

Related to #231 

`Co-authored-by: Evelio Ramos <44717805+eveliososa@users.noreply.github.com>`

### CODE REVIEW

File: src/tests/entities/team.test.ts [hunk 1]
```diff
@@ -1,63 +1,147 @@
...
+const mockQueryBuilder = {
+  where: jest.fn().mockReturnThis(),
+  clone: jest.fn().mockReturnThis(),
+  andWhere: jest.fn().mockReturnThis(),
+  orderBy: jest.fn().mockReturnThis(),
+  addOrderBy: jest.fn().mockReturnThis(),
+  getOne: jest.fn().mockReturnThis(),
+};
+
+describe('', () => {
```
Comment (SpencerKaiser) [inline src/tests/entities/team.test.ts:40]: We need to pass a string in here to describe what we're testing. Looks like the previous one was `judging`... should we add that back here?

File: src/tests/entities/team.test.ts [hunk 2]
```diff
@@ -1,63 +1,147 @@
...
+    team.activeJudgeCount = 2;
+    mockCreateQueryBuilder.execute = jest.fn(() => {
+      team.activeJudgeCount = 1;
+    });
     await team.decrementActiveJudgeCount();
-    const updatedTeam = await Team.findOneOrFail(team.id);
-    expect(updatedTeam.activeJudgeCount).toEqual(0);
+    expect(createQuerySpy).toBeCalledTimes(1);
+    expect(createQuerySpy).toBeCalledWith('team');
+    expect(team.activeJudgeCount).toEqual(1);
```
Comment (SpencerKaiser) [inline src/tests/entities/team.test.ts:56]: This really isn't accomplishing much... your `expect`s here are mostly just checking that the query builder was created for the right entity and that your mock method returns the value you asked it to return. Instead, we should be checking other things like `Was the set method called with an object that includes a method with that appropriate name that attempts to decrement the count?`.

File: src/tests/entities/team.test.ts [hunk 3]
```diff
@@ -1,63 +1,147 @@
...
     await team.incrementJudgeVisits();
-    let updatedTeam = await Team.findOneOrFail(team.id);
-    expect(updatedTeam.judgeVisits).toEqual(1);
-
-    await team.incrementJudgeVisits();
-    updatedTeam = await Team.findOneOrFail(team.id);
-    expect(updatedTeam.judgeVisits).toEqual(2);
+    expect(createQuerySpy).toBeCalledTimes(1);
+    expect(createQuerySpy).toBeCalledWith('team');
+    expect(team.judgeVisits).toEqual(2);
```
Comment (SpencerKaiser) [inline src/tests/entities/team.test.ts:67]: Same comment as the test above;

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
I think many of the tests aren't testing the functionality and are instead testing the test itself... if you want to see if your test was written well, change the implementation significantly and see if the test break. For example, if you modify this line:
``` typescript
      .set({ activeJudgeCount: () => '"team"."activeJudgeCount" - 1' })
```

and change the decrement to 12 (`'"team"."activeJudgeCount" - 1'`), your tests would still pass...

Take another pass, fix the lint errors, and let us know when you're ready for a new review 🙂

File: src/entities/team.ts [hunk 4]
```diff
@@ -95,6 +88,7 @@ export class Team extends BaseEntity {
     throw new Error('Unable to retrieve a team due to concurrency issues');
   }
 
+  /* istanbul ignore next */
   static async updateSelectedTeam(team: Team, hash: string): Promise<UpdateResult> {
```
Comment (SpencerKaiser) [inline src/entities/team.ts:92]: @johnkahn I realized that 99% of these methods are just sql... I guess we could check for the values the use, but I'm not sure I see much value in that. Thoughts?

**Review by johnkahn — APPROVED** [review]

**Pull request merged**